### PR TITLE
[202] Serialize rich text page links

### PR DESCRIPTION
**Pull request by tm-kn**
https://projects.torchbox.com/projects/tbxcom-redesign/tickets/202

Pass slug, service slug and page type so that the links on the Gatsby front-end can be created.

It is required by front-end - torchbox/torchbox-frontend#91

### CODE REVIEW

**Comment by tm-kn** [comment]
Question for later: How do I use document.createElement with GatsbyJS? :confused:

File: tbx/graphql/streamfield.py [hunk 1]
```diff
@@ -66,3 +68,38 @@ def serialise_block(self, block, value):
...
+                            'data-page-slug': page.slug,
+                            'data-page-service-slug': getattr(
+                                getattr(page, 'service', None), 'slug', None
+                            )
+                        }
+                    )
+                    new_tag.append(*anchor.contents)
+                    anchor.replace_with(new_tag)
+                except Page.DoesNotExist:
+                    new_tag = soup.new_tag('a')
```
Comment (zerolab) [inline tbx/graphql/streamfield.py]: Should this not remove the link altogether?

File: tbx/graphql/streamfield.py [hunk 2]
```diff
@@ -66,3 +68,38 @@ def serialise_block(self, block, value):
             return self.serialise_list_block(block, value)
         elif isinstance(block, blocks.StreamBlock):
             return self.serialise_stream_block(block, value)
+
+    def serialize_rich_text(self, source):
+        try:
+            from bs4 import BeautifulSoup
+        except ImportError:
+            return str(RichText(source))
```
Comment (zerolab) [inline tbx/graphql/streamfield.py]: worth commenting why using `RichText()` here where previously it was just returning the source (well, text)

File: tbx/graphql/streamfield.py (hunk 1 above)
Comment (tm-kn) [reply tbx/graphql/streamfield.py]: I think that's default Wagtail behaviour. If the link is in the paragraph, the sentence may look stupid.

Comment (zerolab) [reply tbx/graphql/streamfield.py]: fair enough

**Comment by tm-kn** [comment]
@zerolab https://www.python.org/dev/peps/pep-0008/#maximum-line-length Following PEP8 but can change if you don't like it.

**Comment by zerolab** [comment]
Won't nitpick more. LGTM

**Pull request merged**